py_linq/py_linq.py

```diff
@@ -86,6 +86,15 @@ def select(self, func=lambda x: x):
         """
         return SelectEnumerable(Enumerable(iter(self)), func)
 
+    def for_each(self, func=lambda x: x):
```

Comment:
Looking at this code, it would be an *executing* function when it probably shouldn't be as there is no need. This would have to be tested, but I think this function could just be a wrapper around the output from the `select` function. For example:

```
def for_each(self, func=lambda x: x):
    return self.select(func)
```

Reply:
That was intentional, basically replicating this where input is Action<T> and return is void: https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.list-1.foreach?view=net-6.0